src/plugins/orangehrmClaimPlugin/Api/ClaimExpenseAPI.php

```diff
@@ -0,0 +1,332 @@
+     *                 ref="#/components/schemas/Claim-ExpenseModel"
+     *             ),
+     *             @OA\Property(property="meta", type="object")
+     *         )
+     *     ),
+     *     @OA\Response(response="404", ref="#/components/responses/RecordNotFound")
+     * )
+     *
+     * @inheritDoc
+     */
+    public function getOne(): EndpointResult
+    {
+        $claimExpenseId = $this->getRequestParams()->getInt(RequestParams::PARAM_TYPE_ATTRIBUTE, CommonParams::PARAMETER_ID);
+        $claimExpense = $this->getClaimService()->getClaimDao()->getClaimExpenseById($claimExpenseId);
+        $this->throwRecordNotFoundExceptionIfNotExist($claimExpense, ClaimExpense::class);
+        return new EndpointResourceResult(ClaimExpenseModel::class, $claimExpense);
+    }
+
+    public function getValidationRuleForGetOne(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            $this->getValidationDecorator()->requiredParamRule(
+                new ParamRule(
+                    CommonParams::PARAMETER_ID,
+                    new Rule(Rules::POSITIVE)
+                )
+            ),
+        );
+    }
+
+    /**
+     * @OA\Put(
+     *     path="/api/v2/claim/expenses/{id}",
+     *     tags={"Claim/Expenses"},
+     *     @OA\PathParameter(
+     *         name="id",
+     *         @OA\Schema(type="integer")
+     *     ),
+     *     @OA\RequestBody(
+     *         required=true,
+     *         @OA\JsonContent(
+     *             @OA\Property(property="expense_type_id", type="integer"),
+     *             @OA\Property(property="date", type="string"),
+     *             @OA\Property(property="amount", type="float"),
+     *             @OA\Property(property="note", type="string")
+     *         )
+     *     ),
+     *     @OA\Response(
+     *         response="200",
+     *         description="Success",
+     *         @OA\JsonContent(
+     *             @OA\Property(
+     *                 property="data",
+     *                 ref="#/components/schemas/Claim-ExpenseModel"
+     *             ),
+     *             @OA\Property(property="meta", type="object")
+     *         )
+     *     ),
+     *     @OA\Response(response="404", ref="#/components/responses/RecordNotFound")
+     * )
+     * @inheritDoc
+     */
+    public function update(): EndpointResult
+    {
+        $claimExpenseId = $this->getRequestParams()->getInt(RequestParams::PARAM_TYPE_ATTRIBUTE, CommonParams::PARAMETER_ID);
+        $claimExpense = $this->getClaimService()->getClaimDao()->getClaimExpenseById($claimExpenseId);
+        $this->throwRecordNotFoundExceptionIfNotExist($claimExpense, ClaimExpense::class);
+        $status = $this->getClaimService()->getClaimDao()->getClaimRequestById($claimExpense->getClaimRequest()->getId())->getStatus();
+        if (!$status == ClaimRequest::REQUEST_STATUS_INITIATED) {
+            throw new BadRequestException('Claim request is already approved');
```

Comment:
**Users shall be able to edit expenses to their own claim request when it is in either Initiated or Rejected state**
**Supervisor or Admin user shall be able to Edit expenses for their accessible employee’s claim request when the claim request in Initiated state**